src/main/src/test/java/org/geoserver/security/impl/AbstractAuthorizationTest.java

```diff
@@ -170,12 +195,30 @@ public void setUp() throws Exception {
         layerGroupGlobal = buildLayerGroup("layerGroup", pointStyle, null, arcGridLayer);
         layerGroupTopp = buildLayerGroup("layerGroupTopp", lineStyle, toppWs, statesLayer);
         layerGroupWithSomeLockedLayer = buildLayerGroup("layerGroupWithSomeLockedLayer", lineStyle, toppWs, statesLayer, roadsLayer);
+        
+        // container groups for testing group security
+        namedTreeA = buildLayerGroup("namedTreeA", Mode.NAMED, null, null, null, statesLayer, roadsLayer);
+        nestedContainerE = buildLayerGroup("nestedContainerE", Mode.CONTAINER, null, null, null, forestsLayer);
+        containerTreeB = buildLayerGroup("containerTreeB", Mode.CONTAINER, null, null, null, roadsLayer, landmarksLayer, nestedContainerE);
+        singleGroupC = buildLayerGroup("singleGroupC", Mode.SINGLE, null, null, null, statesLayer, basesLayer);
+        wsContainerD = buildLayerGroup("wsContainerD", Mode.CONTAINER, null, null, nurcWs, arcGridLayer);
 
         // cascaded WMS layer
         cascadedLayer = buildLayer("cascaded", toppWs, WMSLayerInfo.class);
         cascaded = (WMSLayerInfo) cascadedLayer.getResource();
     }
 
+    @After
+    public void cleanupRequestThreadLocal() throws Exception {
+        Dispatcher.REQUEST.remove();
+    }
+    
+    protected void setupRequesThreadLocal(String service) {
```

Comment:
Rename `setupRequesThreadLocal` to `setupRequestThreadLocal`. Could this do both WFS and WMS service and have a @Before?